pydoctor/sphinx_ext/api_output.py

```diff
@@ -0,0 +1,69 @@
+"""
+Generate the API docs using pydoctor to be integrated into Sphinx build system.
+
+Inside the Sphinx conf.py file you need to define the following configuration options:
+
+* pydoctor_cwd: Absolute path in which the pydoctor build is trigerred.
+* pydoctor_args: A list with all the pydoctor command line arguments used to trigger the build.
+
+You must call pydoctor with `--quiet` argument
+as otherwise any extra output is converted into Sphinx warnings.
+"""
+import os
+from sphinx.util import logging
+from sphinx.errors import ConfigError
+
+from contextlib import redirect_stdout
+from io import StringIO
+from pydoctor import __version__
+from pydoctor.driver import main
+
+logger = logging.getLogger(__name__)
+
+
+def on_build_finished(app, exception):
```

Comment:
I'd really like to have type annotations on all new code. Prod me on IRC if you have trouble figuring out the right annotations.

Reply:
Is there a way to run mypy on the diff code and complain if the code has missing / invalid type annotations ?

Something like `diff_cover` but for mypy ? :)

Reply:
Not that I know of. What I want to do in the long term is set `disallow_untyped_defs=True`, so mypy will flag missing annotations as errors. But we're still some time away from that being an option for the whole code base.

We could add that option though and then exempt existing modules that don't have full annotations. That would at least force annotations to be added in new modules and in new code in modules that are fully annotated already.

Reply:
As of #274, `disallow_untyped_defs=True` is now the default; several modules have overrides to turn it off, but at least new modules will have mandatory annotations from the start.